src/core/inductive.ml

```diff
@@ -55,17 +56,74 @@ let gen_ind_typ_codom : popt -> sym -> (tbox list -> tbox) -> string -> tbox =
     | _ -> fatal pos "The type of %a is not supported" pp_symbol ind_sym
   in aux [] !(ind_sym.sym_type)
 
+module T =
+  struct
+
+    (** [fold_app a [b1;...;bn]] returns (... ((a b1) b2) ...) bn. *)
+    let fold_app  : tbox -> tbox list -> tbox = List.fold_left _Appl
+
+    (** [fold_app_special f ts] returns the same result that
+        fold_app (_Symb [f]) [ts]. *)
+    let fold_app_special : sym -> tbox list -> tbox = fun f ts ->
+      fold_app (_Symb f) ts
+
+    (** [prf_of p c ts t] returns a proof of the form
+        c.symb_prf ( (((p ts1) ...) tsn) t) where ts = [ts1;...;tsn]. *)
+    let prf_of : tvar -> config -> tbox list -> tbox -> tbox = fun p c ts t ->
+      fold_app_special c.symb_prf [_Appl (fold_app (_Vari p) ts) t]
+  end
+
+(** [preprocessing pos c ind_typ_list p_str x_str] creates an associated list
+    between an inductive type and its predicat, and the list of conclusions,
```

Comment:
predicat -> predicate

Reply:
associated -> association

Reply:
The comment is not clear. This function computes elements of the induction principles that we want to associate to a list of mutually defined inductive types. It returns a pair of lists. The first list is an association list between inductive types and predicate variables (but why do you return both a tvar and a tbox?). The second list is the conclusions of the induction principles.
Prior to that, at the beginning of the file, you should provide a general explanation of the induction principles you want to associate to a list of mutually defined inductive types. In particular, note that you currently generate an induction principle for each inductive type. It is important to mention this as one could expect a single induction principle for the whole list of inductive types by using a conjunction operator /\:Prop->Prop->Prop. By the way, what is best?

Reply:
You should also add that p_str is used as prefix for predicate variables, and x_str as prefix for type parameters.

Reply:
The first list is an association list mapping every inductive type to a pair made of a predicate variable and its type.

Reply:
> The first list is an association list between inductive types and predicate variables (but why do you return both a tvar and a tbox?). 

The name of a predicate variable has the type tvar and its type has the type tbox.

Reply:
>  By the way, what is best?

The current solution if you would like that the user defines only the minimal set of operators.

Reply:
Right. But conjunction is already a required builtin for the why3 tactic. For the moment, we can stick to the current solution. But a deeper reflection is needed, based on practice. You should have a look at proofs on mutually defined inductive types to see what is best in practice.